server/core/src/main/java/dev/slimevr/osc/VMCHandler.kt

```diff
@@ -346,53 +361,53 @@ class VMCHandler(
 						)
 
 					for (unityBone in UnityBone.entries) {
-						val boneType = unityBone.boneType ?: continue
-						// Get SlimeVR bone
-						val bone = humanPoseManager.getBone(boneType)
-
-						// Update unity hierarchy from bone's global
-						// rotation
-						outputUnityArmature
-							?.setGlobalRotationForBone(
-								unityBone,
-								bone!!.getGlobalRotation() * bone.rotationOffset.inv(),
-							)
+						val boneType = unityBone.boneType
+						if (boneType != null) {
+							// Get SlimeVR bone
+							val bone = humanPoseManager.getBone(boneType)
+
+							// Update unity hierarchy from bone's global
+							// rotation
+							outputUnityArmature!!
+								.setGlobalRotationForBone(
+									unityBone,
+									bone.getGlobalRotation().times(bone.rotationOffset.inv()),
+								)
+						}
```

Comment:
WHY IS THE CODE GETTING WORST
```suggestion
						val boneType = unityBone.boneType ?: continue
						// Get SlimeVR bone
						val bone = humanPoseManager.getBone(boneType)

						// Update unity hierarchy from bone's global
						// rotation
						outputUnityArmature!!
							.setGlobalRotationForBone(
								unityBone,
								bone.getGlobalRotation() * bone.rotationOffset.inv(),
							)
```